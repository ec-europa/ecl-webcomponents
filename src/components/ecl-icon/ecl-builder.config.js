const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const isProd = process.env.NODE_ENV === 'production';
const outputFolder = path.resolve(__dirname, isProd ? './dist' : './build');

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-icon/icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-icon/icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
