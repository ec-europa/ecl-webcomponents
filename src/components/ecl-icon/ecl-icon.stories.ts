import iconPathEc from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
import iconPathEu from '@ecl/resources-eu-icons/dist/sprites/icons.svg';
import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

export default {
  title: 'Components/icon',
};

const Template = (args) => `<ecl-icon 
                              icon="${args.icon}"
                              color="${args.color}"
                              size="${args.size}"
                              transform="${args.transform}"
                              path="${args.theme === 'ec' ? iconPathEc : iconPathEu}"
                            ></ecl-icon>`;

export const Icon = Template.bind({});
Icon.storyName = 'default';
Icon.args = {
  theme: 'ec',
  icon: 'close',
  transform: '',
  color: 'default',
  size: '2xl',
};
Icon.argTypes = {
  theme: {
    type: { name: 'select'},
    options: ['ec', 'eu'],
  },
  icon: {
    type: { name: 'select' },
    options: iconsAllEc,
    description: "Add an icon",
  },
  size: {
    type: { name: 'select' },
    options: ['2xs','xs', 's', 'm', 'l', 'xl', '2xl'],
    description: "icon size",
  },
  color: {
    type: { name: 'select' },
    options: ['default', 'primary', 'inverted'],
    description: "icon color",
  },
  transform: {
    type: { name: 'select' },
    description: 'Button icon transform',
    options: [
      'rotate-90',
      'rotate-180',
      'rotate-270',
      'flip-horizontal',
      'flip-vertical',
    ],
  },
};