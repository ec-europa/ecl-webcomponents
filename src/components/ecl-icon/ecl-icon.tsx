import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}