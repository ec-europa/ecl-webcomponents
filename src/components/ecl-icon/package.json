{
  "name": "@ecl/web-components-icon",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "3.5.0",
  "description": "ECL Icon",
  "main": "ecl-icon.ts",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-p clean:build build:*",
    "clean": "run-s clean:*",
    "clean:build": "rimraf build",
    "clean:dist": "rimraf dist",
    "dist": "cross-env NODE_ENV=production run-p clean:dist build:*",
    "prepublish": "npm run clean && npm run build && npm run dist"
  },
  "devDependencies": {
    "@ecl/builder": "3.5.0",
    "@ecl/specs-component-icon": "3.5.0",
    "@ecl/vanilla-component-icon": "3.5.0",
    "@ecl/resources-ec-icons": "3.5.0",
    "@ecl/resources-eu-icons": "3.5.0"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}