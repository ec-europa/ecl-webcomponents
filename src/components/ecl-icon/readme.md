# ecl-icon

<!-- Auto Generated Below -->


## Properties

| Property     | Attribute     | Description | Type     | Default     |
| ------------ | ------------- | ----------- | -------- | ----------- |
| `color`      | `color`       |             | `string` | `undefined` |
| `icon`       | `icon`        |             | `string` | `undefined` |
| `path`       | `path`        |             | `string` | `undefined` |
| `size`       | `size`        |             | `string` | `'xs'`      |
| `styleClass` | `style-class` |             | `string` | `undefined` |
| `theme`      | `theme`       |             | `string` | `'ec'`      |
| `transform`  | `transform`   |             | `string` | `undefined` |


## Dependencies

### Used by

 - [ecl-accordion-item](../ecl-accordion)
 - [ecl-expandable](../ecl-expandable)

### Graph
```mermaid
graph TD;
  ecl-accordion-item --> ecl-icon
  ecl-expandable --> ecl-icon
  style ecl-icon fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
