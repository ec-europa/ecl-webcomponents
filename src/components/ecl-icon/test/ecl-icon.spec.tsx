import { newSpecPage } from '@stencil/core/testing';
import { EclIcon } from '../ecl-icon';

describe('ecl-icon', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclIcon],
      html: `<ecl-icon
              icon="close-filled"
              color="primary"
              size="l"
              path="/icons.svg"
             ></ecl-icon>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
