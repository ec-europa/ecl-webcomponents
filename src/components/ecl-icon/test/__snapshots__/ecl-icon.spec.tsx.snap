// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-icon renders 1`] = `
<ecl-icon color="primary" icon="close-filled" path="/icons.svg" size="l">
  <svg class="ecl-icon ecl-icon--l ecl-icon--primary">
    <use xlink:href="/icons.svg#close-filled"></use>
  </svg>
</ecl-icon>
`;
